### PR TITLE
New package: kindd-2.1.0

### PR DESCRIPTION
**Pull request by LinArcX**
A kindful dd gui(front-end) written in qt quick:
https://github.com/LinArcX/kindd

### CODE REVIEW

File: srcpkgs/kindd/template [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+maintainer="linarcx <linarcx@gmail.com>"
+license="GPL-3.0-or-later"
+homepage="https://github.com/LinArcX/Kindd"
+distfiles="${homepage}/archive/${version}.tar.gz"
+checksum=5dbfcbc07cdc36a6a5d2b40ed5e8159a305ccc5608580635961ed9ad6f43e2b3
+
+do_install() {
+	vmkdir usr/bin
+	vmkdir usr/share/pixmaps
+	vmkdir usr/share/applications
```
Comment (Johnnynator) [inline srcpkgs/kindd/template]: vmkdirs not needed

File: srcpkgs/kindd/template [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+	vbin ./kindd 755 usr/bin
```
Comment (Johnnynator) [inline srcpkgs/kindd/template]:
```suggestion
	vbin kindd
```
vbin already sets the permissions and path

**Comment by LinArcX** [comment]
It fails for some ARCHs:
https://travis-ci.org/void-linux/void-packages/builds/590781795?utm_source=github_status&utm_medium=notification
What should i do?

**Comment by Chocimier** [comment]
> It fails for some ARCHs: What should i do?

Qmake often fails to cross-compile. One way is to add some makedepends to hostmakedepends, other is to add not detected include paths like in `qt5-fsarchiver` template.

> Files of kindd:
> /usr/bin/755

Test your packages before submission, or leave comment that they are untested/wip.

> => kindd-2.0.1_1: running pre-pkg hook: 04-generate-runtime-deps ...
>
>   SONAME: libQt5QuickControls2.so.5 <-> qt5-quickcontrols2>=5.7.1_1

xbps-src detects what libraries package do link with, so no need to define most of libraries in depends.

> wrksrc="Kindd-${version}"

According to [Manual.md](https://github.com/void-linux/void-packages/blob/master/Manual.md#programs), Kindd is better choice for pkgname then.

> fatal: not a git repository (or any of the parent directories): .git

Can you check if it builds without git in hostmakedepends?

File: srcpkgs/kindd/template [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+version=2.0.1
+revision=1
+wrksrc="Kindd-${version}"
+build_style=qmake
+hostmakedepends="git qt5-qmake"
+makedepends="qt5-devel qt5-declarative-devel qt5-quickcontrols2-devel"
+depends="polkit-gnome qt5-quickcontrols qt5-graphicaleffects qt5-quickcontrols2 qt5-declarative qt5-svg"
+short_desc="Kindful dd gui written in qt quick"
+maintainer="linarcx <linarcx@gmail.com>"
+license="GPL-3.0-or-later"
```
Comment (Chocimier) [inline srcpkgs/kindd/template]: GPL-3.0-only

Comment (LinArcX) [reply srcpkgs/kindd/template]: why?

**Comment by LinArcX** [comment]
@Chocimier 
1. Can i set specific ARCHs for this package? (For example said, only build for x86_64 and i686)?
2. I don't have this file: `/usr/bin/755`.
3. `so no need to define most of libraries in depends`. Exactly which of them should i remove?
4. I will rename the project to __kindd__, and there is no wrksrc needed anymore after that.
5. I'll check it tomorow.

File: srcpkgs/kindd/template (hunk 3 above)
Comment (Chocimier) [reply srcpkgs/kindd/template]: From LICENSE:
> Each version is given a distinguishing version number.  If the
Program specifies that a certain numbered version of the GNU General
Public License "or any later version" applies to it, you have the
option of following the terms and conditions either of that numbered
version or of any later version published by the Free Software
Foundation.  If the Program does not specify a version number of the
GNU General Public License, you may choose any version ever published
by the Free Software Foundation.

LICENSE is GPL-3.0 so it specifies version, but I can't find permission for later ones.

**Comment by Johnnynator** [comment]
>     2. I don't have this file: `/usr/bin/755`.

You have, the wrong usage vbin I marked renamed your binary to 755

**Comment by LinArcX** [comment]
> > ```
> > 2. I don't have this file: `/usr/bin/755`.
> > ```
> 
> You have, the wrong usage vbin I marked renamed your binary to 755

I changed 
`vbin ./kindd 755 usr/bin`
to:
`vbin kindd`
in new version. It's correct, isn't it?

**Comment by Chocimier** [comment]
>     1. Can i set specific ARCHs for this package? (For example said, only build for x86_64 and i686)?

Archs is for marking packages that are strictly tied to arch, eg. written in assembler. Most of qmake-built packages do cross build with small effort.

>     3. `so no need to define most of libraries in depends`. Exactly which of them should i remove?

Detected by xbps-src, see output of build, I quoted what to look for already.

> It's correct, isn't it?

You can check it.

**Comment by LinArcX** [comment]
> > ```
> > 1. Can i set specific ARCHs for this package? (For example said, only build for x86_64 and i686)?
> > ```
> 
> Archs is for marking packages that are strictly tied to arch, eg. written in assembler. Most of qmake-built packages do cross build with small effort.
> 
> > ```
> > 3. `so no need to define most of libraries in depends`. Exactly which of them should i remove?
> > ```
> 
> Detected by xbps-src, see output of build, I quoted what to look for already.
> 
> > It's correct, isn't it?
> 
> You can check it.

No, no. I mean just build this package for x86_64 and i686. I see some options like nocross:
`nocross="kaccounts-integration is nocross"`
or
archs="x86_64*i686*"
Can i use them?

**Comment by LinArcX** [comment]
@Chocimier, @Johnnynator 
I changed build-tools from qmake to cmake and update my repo and this template file.

**Comment by LinArcX** [comment]
Still i don't understand [why Qt5::Core should be depned on qmake](https://travis-ci.org/github/void-linux/void-packages/jobs/662470362#L648)?

I totally removed qmake and replaced it with cmake. Why shouldn't be compiled on arm arch?

File: srcpkgs/kindd/template [hunk 4]
```diff
@@ -0,0 +1,21 @@
+# Template file for 'kindd'
+pkgname=kindd
+version=2.1.0
+revision=1
+wrksrc="kindd-${version}"
+build_style=cmake
+hostmakedepends="cmake pkg-config"
```
Comment (Johnnynator) [inline srcpkgs/kindd/template]:
```suggestion
hostmakedepends="cmake pkg-config qt5-qmake"
```
cmake just likes to use qmake to detect some stuff, idk what exactly it does with it out of my head.

File: srcpkgs/kindd/template [hunk 5]
```diff
@@ -0,0 +1,21 @@
+# Template file for 'kindd'
+pkgname=kindd
+version=2.1.0
+revision=1
+wrksrc="kindd-${version}"
```
Comment (Johnnynator) [inline srcpkgs/kindd/template]: Just remove this line, this is the default value.

Comment (LinArcX) [reply srcpkgs/kindd/template]: Done!

File: srcpkgs/kindd/template (hunk 4 above)
Comment (LinArcX) [reply srcpkgs/kindd/template]: Yeah, i've also added `qt5-host-tools` due to some complains that arm ARCH about:
>  The imported target "Qt5::Core" references the file

     "/usr/lib/qt5/bin/moc"

**Comment by LinArcX** [comment]
@Johnnynator Seems it's not still in repo. Why?

**Comment by Johnnynator** [comment]
The build server is busy building cross toolchains. So it might take a few hours.

**Pull request merged**